### PR TITLE
TST: Test renaming a multiindex with duplicates (GH 38015)

### LINKED ISSUE #38015: BUG: df.rename raises when multiindex has duplicates
```python
>>> idx = pd.Index([('A','cat'),('B','cat'),('B','cat')])
>>> df = pd.DataFrame(index=idx)
>>> df.rename(index={'A':'Apple'}, level=0)
ValueError: non-broadcastable output operand with shape (1,1) doesn't match the broadcast shape (1,2)
```

### PR DESCRIPTION
**Pull request by andrewwoood**
- [x] closes #38015
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

This is my first real contribution, I appreciate the feedback!

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_multiindex.py [hunk 1]
```diff
@@ -98,3 +98,13 @@ def test_multiindex_with_datatime_level_preserves_freq(self):
         result = df.loc[0].index
         tm.assert_index_equal(result, dti)
         assert result.freq == dti.freq
+
+    def test_rename_multiindex_with_duplicates(self):
+        # GH 38015
+        idx = pd.Index([("A", "cat"), ("B", "cat"), ("B", "cat")])
```
Comment (phofl) [inline pandas/tests/indexing/multiindex/test_multiindex.py:104]: Looks good in General. Please use the MultiIndex constructor and compare the whole dataframe

**Comment by phofl** [comment]
Some pre-commit issues, otherwise lgtm. 

You could install pre-commit to run the hooks locally before pushing

**Comment by jreback** [comment]
can you merge master and ping on green

**Comment by andrewwoood** [comment]
@jreback Sorry, is this what you were asking me to do? I'm not very competent with using git and contributing to open-source quite yet. I tried using the pre-commit framework but it doesn't seem to help my file pass the pre-commit check.

**Comment by phofl** [comment]
The ci tells you whats wrong:

```
 pandas/tests/indexing/multiindex/test_multiindex.py:125:14: PDF019 found both 'pd.MultiIndex' and 'MultiIndex' in the same file
pandas/tests/indexing/multiindex/test_multiindex.py:126:14: PDF019 found both 'pd.DataFrame' and 'DataFrame' in the same file
pandas/tests/indexing/multiindex/test_multiindex.py:129:15: PDF019 found both 'pd.MultiIndex' and 'MultiIndex' in the same file
pandas/tests/indexing/multiindex/test_multiindex.py:130:20: PDF019 found both 'pd.DataFrame' and 'DataFrame' in the same file
```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @andrewwoood! Happy to have you address similar `good first issue`s that need tests.

**Pull request merged**